--- IxLoad.py.org	2018-08-18 23:34:50.840075631 +0900
+++ IxLoad.py	2018-08-28 13:26:34.644063184 +0900
@@ -14,16 +14,16 @@
 #
 ################################################################################
 
-import Tkinter, sys, os, re
+import tkinter, sys, os, re
 
 
 try:
     # create a TCL interpreter instance (Python 2.5+)
-    _tcl_ = Tkinter.Tcl()
+    _tcl_ = tkinter.Tcl()
     _tclCallback_ = _tcl_
 except:
     # create a TCL interpreter instance (Python 2.4-)
-    _tk_ = Tkinter.Tk()
+    _tk_ = tkinter.Tk()
     _tcl_ = _tk_.tk
     _tclCallback_ = _tk_
 
@@ -72,7 +72,7 @@ def _python2Tcl(value, prefix = ""):
     result = None
     if isinstance(value, IxLoadObjectProxy):
         result = " %s%s" % (prefix, value._tclObj_)
-    elif isinstance(value, basestring):
+    elif isinstance(value, str):
         # scriptgen sometimes generates python with escaped brackets [] which
         # isn't necessary for python, so here we need to escape them only if not already escaped
         # Avoid having a path formatted like this: ""[path]""
